src/lib/planviz/lqp_visualizer.cpp

```diff
@@ -82,11 +98,16 @@ void LQPVisualizer::_build_dataflow(const std::shared_ptr<AbstractLQPNode>& from
     label_stream << "no est.";
   }
 
-  VizEdgeInfo info = _default_edge;
-  info.label = label_stream.str();
+  VizEdgeInfo info = edge_info;
+  info.label = edge_info.label.empty() ? label_stream.str() : edge_info.label;
```

Comment:
This is a hacky (and therefore hard to read) way to introduce label overrides. You're calculating all this (somewhat nonsense) info that goes into label_stream, only to discard it here.